pkg/goDB/engine/query_test.go

```diff
@@ -125,51 +126,96 @@ func TestSimpleQuery(t *testing.T) {
 	}
 }
 
-func TestInterfaceValidation(t *testing.T) {
+type MockInterfaceLister struct{}
 
-	// create args
-	var tests = []struct {
-		iface       string
-		expectedErr error
-	}{
+func (mockLister MockInterfaceLister) GetInterfaces(dbPath string) ([]string, error) {
+	return info.GetInterfaces(dbPath)
+}
+
+type InterfaceListerFunc func(dbPath string) ([]string, error)
+
+func (f InterfaceListerFunc) GetInterfaces(dbPath string) ([]string, error) {
+	return f(dbPath)
+}
+
+type filteringTestDefinition struct {
+	name     string
+	argument string
+	ifaces   []string
+	expected []string
+}
+
+func TestCommaSeparatedInterfaceFiltering(t *testing.T) {
+	var tests = []filteringTestDefinition{
 		{
-			"",
-			errors.New("interface list contains empty interface name"),
+			"selected interfaces are returned",
+			"eth0, eth2",
+			[]string{"eth0", "wlan0", "eth2"},
+			[]string{"eth0", "eth2"},
 		},
 		{
-			"eth/0",
-			errors.New("interface name `eth/0` is invalid"),
+			"nonexistent interface is ignored",
+			"eth0, eth2, notexistent",
+			[]string{"eth0", "wlan0", "eth2"},
+			[]string{"eth0", "eth2"},
 		},
 		{
-			"eth 0",
-			errors.New("interface name `eth 0` is invalid"),
+			"any arguments returns all ignored",
+			"any",
+			[]string{"eth0", "wlan0", "eth2"},
+			[]string{"eth0", "wlan0", "eth2"},
 		},
 		{
-			"thisinterfacenameisfartoolongtobesupported",
-			errors.New("interface name `thisinterfacenameisfartoolongtobesupported` is invalid"),
+			"all but negaded interface is returned",
+			"any,!eth2,!t1",
+			[]string{"eth0", "wlan0", "t1", "t2", "eth2"},
+			[]string{"eth0", "wlan0", "t2"},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			lister := InterfaceListerFunc(func(dbPath string) ([]string, error) {
```

Comment:
Nice idea to go functional. But unnecessary if you remove the `dbPath` from the interface's description.